token-lending/program/src/pyth.rs

```diff
@@ -0,0 +1,148 @@
+#![allow(missing_docs)]
+
+use bytemuck::{
+    cast_slice, cast_slice_mut, from_bytes, from_bytes_mut, try_cast_slice, try_cast_slice_mut,
+    Pod, PodCastError, Zeroable,
+};
+use std::mem::size_of;
+
+pub const MAGIC: u32 = 0xa1b2c3d4;
+pub const VERSION_1: u32 = 1;
+pub const VERSION: u32 = VERSION_1;
+pub const MAP_TABLE_SIZE: usize = 640;
+pub const PROD_ACCT_SIZE: usize = 512;
+pub const PROD_HDR_SIZE: usize = 48;
+pub const PROD_ATTR_SIZE: usize = PROD_ACCT_SIZE - PROD_HDR_SIZE;
+
+#[derive(Default, Copy, Clone)]
+#[repr(C)]
+pub struct AccKey {
+    pub val: [u8; 32],
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub enum AccountType {
+    Unknown,
+    Mapping,
+    Product,
+    Price,
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub enum PriceStatus {
+    Unknown,
+    Trading,
+    Halted,
+    Auction,
+}
+
+impl Default for PriceStatus {
+    fn default() -> Self {
+        PriceStatus::Trading
+    }
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub enum CorpAction {
+    NoCorpAct,
+}
+
+impl Default for CorpAction {
+    fn default() -> Self {
+        CorpAction::NoCorpAct
+    }
+}
+
+#[derive(Default, Copy, Clone)]
+#[repr(C)]
+pub struct PriceInfo {
+    pub price: i64,
+    pub conf: u64,
+    pub status: PriceStatus,
+    pub corp_act: CorpAction,
+    pub pub_slot: u64,
+}
+
+#[derive(Default, Copy, Clone)]
+#[repr(C)]
+pub struct PriceComp {
+    publisher: AccKey,
+    agg: PriceInfo,
+    latest: PriceInfo,
+}
+
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub enum PriceType {
+    Unknown,
+    Price,
+    #[allow(clippy::upper_case_acronyms)]
+    TWAP,
+    Volatility,
+}
+
+impl Default for PriceType {
+    fn default() -> Self {
+        PriceType::Price
+    }
+}
+
+#[derive(Default, Copy, Clone)]
```

Comment:
nit: I don't think the `Default` auto derive here is helpful

Reply:
True, this was copied from an example of an Anchor program using Pyth. I'll clean these up.

Reply:
Removed, none of the defaults were needed.